pppoe/discovery.go

```diff
@@ -224,6 +224,36 @@ func parsePADS(buf []byte) (sessionID uint16, err error) {
 	return pkt.SessionID, nil
 }
 
+func readPADT(conn net.PacketConn, concentrator net.HardwareAddr, sessionID uint16) error {
+	var b [1500]byte
+
+	for {
+		n, from, err := conn.ReadFrom(b[:])
+		if err != nil {
+			return err
+		}
+
+		if concentrator.String() != from.String() {
+			// Wrong peer, keep waiting
+			continue
+		}
+
+		pkt, err := parseDiscoveryPacket(b[:n])
+		if err != nil {
+			// Bad packet, keep waiting
```

Comment:
Seems a bit weird to just discard this error and not even log it.

Perhaps readPADT should be a method on something with configurable log policy. TODO to let user get at the error somehow?

Or comment about why the error can't be interesting?